cache/cache.go

```diff
@@ -97,6 +100,33 @@ func (r *RedisClient) GetBool(key string) (bool, error) {
 	return strconv.ParseBool(result)
 }
 
+// SetInt saves an integer value to REDIS.
+// If the client was set up with a prefix it will be added in front of the key.
+// Zero expiration means the key has no expiration time.
+func (r *RedisClient) SetInt(key string, value int64, expiration time.Duration) error {
+	return r.Set(key, strconv.FormatInt(value, 10), expiration)
+}
+
+// GetInt retrieves an integer value from REDIS.
+// If the client was set up with a prefix it will be added in front of the key.
+func (r *RedisClient) GetInt(key string) (int64, error) {
+	result, err := r.Get(key)
+	if err == redis.Nil {
+		return 0, ErrNotFound
+	} else if err != nil {
+		return 0, err
+	}
+
+	return strconv.ParseInt(result, 10, 64)
+}
+
+// Incr increments a value in REDIS.
+// If the client was set up with a prefix it will be added in front of the key.
+// It returns the new (incremented) value.
+func (r *RedisClient) Incr(key string) (int64, error) {
+	return r.Redis.Incr(r.prefixedKey(key)).Result()
```

Comment:
Shouldn't we check whether the error is `redis.Nil` here as well and then return `ErrNotFound`?

Reply:
If we call for example `Incr("this_key_doesnt_exists_in_redis_yet")`, the value we find in Redis for it will be 1 (the same you will get if you directly do this in in Redis). I am not sure when `incr` will return `redis.Nil`.